src/Native/System.Native/pal_networking.cpp

```diff
@@ -2321,7 +2336,15 @@ static Error WaitForSocketEventsInner(int32_t port, SocketEvent* buffer, int32_t
         // This copy is made deliberately to avoid overwriting data.
         kevent64_s evt = events[i];
         buffer[i] = {
+#if HAVE_KQUEUE && !HAVE_KEVENT64_S
+            // when kevent64_t is actually kevent
+            // (see the 'typedef kevent' above),
+            // evt.udata is void* and hence static_cast
+            // does not work.
+            .Data = reinterpret_cast<uintptr_t>(evt.udata),
+#else
             .Data = static_cast<uintptr_t>(evt.udata),
+#endif
             .Events = GetSocketEvents(evt.filter, evt.flags)
```

Comment:
@nguerrera we could use `kevent` rather than `kevent64`. I originally used `kevent64` for the managed-only implementation due to its fixed size across bitnesses, but that's less of an issue here.